### ISSUE TITLE
RSS feed with publish dates in the future

### ISSUE THREAD

**Issue by signalstoerung** [signalstoerung/reader#3]
[This feed](https://www.taipeitimes.com/xml/index.rss) generated “published” dates in the future for the entire feed, so the first five pages of results are now stories from this source.

**Comment by signalstoerung** [comment]
A simple check [here](https://github.com/signalstoerung/reader/blob/d4a8c49ebd66aab0fdac843bfe28eeb5ed485ec8/ingest.go#L45) whether PublishedParsed is in the future (and if so, replace it with the current time) should address this.

Bottom line, we shouldn’t blindly trust the date in the feed, it may be the result of a misconfiguration. There’s no use case for Reader with news from the future!

**Comment by signalstoerung** [comment]
I checked the original feed at different times today, and the root cause seems to be a bug on the server that generates the feed.

This morning, the time stamp (on all items, as a matter of fact) was 2023-11-13T08:00:00+08:00, or 00:00 GMT.

This evening, it was 2023-11-14T08:00:00+08:00 — 00:00 GMT tomorrow, still several hours in the future. 

Some error on the server, if I had to guess, converting local time to UTC. 

Why not use the actual publish time of the articles? Shouldn’t that be in the CMS? Why set the publish time to some arbitrary time? We’ll never know. In the meantime, let’s not trust the time of the feed if it’s in the future.
